data_structures/linked_list/reverse_linked_list_program.py

```diff
@@ -0,0 +1,55 @@
+# Time Complexity : O(n) 
+# Space Complexity : O(1) 
+
+# Node class 
+class Node: 
+
+	# Constructor to initialize the node object 
+	def __init__(self, data): 
+		self.data = data 
+		self.next = None
+
+class LinkedList: 
+
+	# Function to initialize head 
+	def __init__(self): 
+		self.head = None
+
+	# Function to reverse the linked list 
+	def reverse(self): 
+		prev = None
+		current = self.head 
+		while(current is not None): 
+			next = current.next
+			current.next = prev 
+			prev = current 
+			current = next
+		self.head = prev 
+		
+	# Function to insert a new node at the beginning 
+	def push(self, new_data): 
+		new_node = Node(new_data) 
+		new_node.next = self.head 
+		self.head = new_node 
+
+	# Utility function to print the linked LinkedList 
+	def printList(self): 
```

Comment:
```suggestion 
	def print_list(self): 
	"""Utility function to print the linked LinkedList"""
```
The function name should be in `snake_case` as discussed in CONTRIBUTING.md

Please add `doctest` and type hints to this function